src/components/functions.js

```diff
@@ -1,10 +1,166 @@
-import axios from 'axios';
-import {BASE_URL,OPTIMIZE_SVG} from '../constants/constants';
-
-export default  function optimizeSvg(dataUrl){
-    return axios.post(`${BASE_URL}${OPTIMIZE_SVG}`, { dataUrl })
-        .then(res => {
-            return res.data;
-        })
-        .catch(err => console.log(err));
+import axios from "axios";
+import React from "react";
+import DisplaySvg from "../components/displaySvg/displaySvg";
+import finalSvgDisplay from "../components/displaySvg/finalSvgDisplay.scss";
+import { BASE_URL, OPTIMIZE_SVG } from "../constants/constants";
+import base64 from "base-64";
+
+export const optimizeSvg = (dataUrl, svgo) => {
+  return axios
+    .post(`${BASE_URL}${OPTIMIZE_SVG}`, { dataUrl, svgo })
+    .then(res => {
+      return res.data.urlData;
+    })
+    .catch(err => console.log(err));
+};
+
+export const immutablePush=(arr, newEntry)=>{
+  return [].concat(arr, newEntry)
 }
```

Comment:
if you wish to push to an array immutable you could also just say [...arr, newEntry] this will spread your existing array into a new one and add in the new entry to it

Reply:
I like this approach, but I am no longer using this function as its have been removed from function.js file and I will you this approach in future. thanks